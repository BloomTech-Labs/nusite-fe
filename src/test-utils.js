import React from "react";
import { render } from "@testing-library/react";
import { ThemeContext } from "./context/contexts";
import UserContext, { ANONYMOUS_USER } from "./context/user/context";

function MutationObserver(callback) {
   this.observe = jest.fn();
   this.disconnect = jest.fn();
   this.takeRecords = jest.fn();
}
global.MutationObserver = jest.fn(MutationObserver);

const mockSpies = {
   setDarkMode: jest.fn(),
   userDispatch: jest.fn(),
};

const AllTheProviders = ({ children }) => (
   <ThemeContext.Provider
      value={{ darkMode: false, setDarkMode: mockSpies.setDarkMode }}
   >
      <UserContext.Provider
         value={{
            userData: ANONYMOUS_USER,
            userDispatch: mockSpies.userDispatch,
         }}
      >
         {children}
      </UserContext.Provider>
   </ThemeContext.Provider>
);

const customRender = (ui, options) => {
   return {
      ...render(ui, { wrapper: AllTheProviders, ...options }),
      mockSpies,
   };
};

const testLinkArray = (linkArray, expected_length, expected_refs) => {
   expect(linkArray.length).toBe(expected_length);
   linkArray.forEach(element => {
      expect(expected_refs).toContain(element.getAttribute("href"));
   });
};

export * from "@testing-library/react";
export { customRender as render, testLinkArray };
