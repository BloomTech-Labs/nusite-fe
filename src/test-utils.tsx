import React from "react";
import { render, RenderResult } from "@testing-library/react";

import { ThemeContext } from "./context/contexts";

interface SpiesMap {
   setDarkMode: jest.Mock;
}

interface RenderWithSpiesResult extends RenderResult {
   mockSpies: SpiesMap;
}

const mockSpies: SpiesMap = {
   setDarkMode: jest.fn(),
};

const AllTheProviders: React.FC = ({ children }) => (
   <ThemeContext.Provider
      value={{ darkMode: false, setDarkMode: mockSpies.setDarkMode }}
   >
      {children}
   </ThemeContext.Provider>
);

const customRender = (ui: React.ReactElement, options?: Object | undefined) => {
   return {
      ...render(ui, { wrapper: AllTheProviders, ...options }),
      mockSpies,
   };
};

export * from "@testing-library/react";
export { customRender as render, RenderWithSpiesResult };
