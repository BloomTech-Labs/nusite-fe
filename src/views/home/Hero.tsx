import React from "react";
import { Grid, makeStyles, useMediaQuery } from "@material-ui/core/";
import useTheme from "@material-ui/core/styles/useTheme";
import heroImage from "../../images/hero.jpeg";

const useStyles = makeStyles(theme => ({
   gridContainer: {
      maxWidth: "1200px",
   },
   heroImage: {
      width: "100%",
   },
   typography: {
      ...theme.typography.h1,
      [theme.breakpoints.down("xl")]: {
         fontSize: "35px",
      },
      [theme.breakpoints.down("lg")]: {
         fontSize: "32px",
      },
      [theme.breakpoints.down("md")]: {
         fontSize: "25px",
      },
      [theme.breakpoints.down("sm")]: {
         fontSize: "18px",
      },
      [theme.breakpoints.down(414)]: {
         fontSize: "16px",
      },
   },
   textContainer: {
      paddingTop: "2.5%",
      display: "flex",
      width: "35%",
      maxWidth: "400px",
      textAlign: "left",
      justifyContent: "left",
      margin: "0 auto",
      position: "absolute",
      left: "60%",
      maxHeight: "80%",
      [theme.breakpoints.down("xs")]: {
         paddingTop: 0,
      },
   },
}));

const Hero = (props: any) => {
   const classes = useStyles();
   const theme = useTheme();
   const matches = useMediaQuery(theme.breakpoints.down("xs"));

   const heroText = (
      <h1 className={classes.typography}>
         Connecting people with great ideas to web developers. Old website, new
         website, app, or feature, we've got you covered.
      </h1>
   );
   const mobileHeroText = (
      <h1 className={classes.typography}>
         Connecting people with great ideas to developers.
      </h1>
   );
   return (
      <Grid item className={classes.gridContainer}>
         <Grid container>
            <Grid container justify="center">
               <Grid item>
                  <img
                     className={classes.heroImage}
                     src={heroImage}
                     alt="man in a desk with a computer extending hand for a handshake"
                  />
               </Grid>
               <Grid item className={classes.textContainer}>
                  {matches ? mobileHeroText : heroText}
               </Grid>
            </Grid>
         </Grid>
      </Grid>
   );
};
export default Hero;
