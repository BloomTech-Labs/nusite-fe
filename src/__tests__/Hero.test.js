import "@testing-library/jest-dom/extend-expect";

import React from "react";
import { render, cleanup, screen } from "../test-utils";

import Hero from "../views/home/Hero";

describe("Hero component tests", () => {
   test("It displays correctly.", () => {
      const hero = render(<Hero />);
      expect(hero).toMatchSnapshot();
   });
});
