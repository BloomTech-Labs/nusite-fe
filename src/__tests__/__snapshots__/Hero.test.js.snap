// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero component tests It displays correctly. 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiGrid-root makeStyles-gridContainer-1 MuiGrid-item"
      >
        <div
          class="MuiGrid-root MuiGrid-container"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
          >
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <img
                alt="man in a desk with a computer extending hand for a handshake"
                class="makeStyles-heroImage-2"
                src="hero.jpeg"
              />
            </div>
            <div
              class="MuiGrid-root makeStyles-textContainer-4 MuiGrid-item"
            >
              <h1
                class="makeStyles-typography-3"
              >
                Connecting people with great ideas to web developers. Old website, new website, app, or feature, we've got you covered.
              </h1>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiGrid-root makeStyles-gridContainer-1 MuiGrid-item"
    >
      <div
        class="MuiGrid-root MuiGrid-container"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
        >
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <img
              alt="man in a desk with a computer extending hand for a handshake"
              class="makeStyles-heroImage-2"
              src="hero.jpeg"
            />
          </div>
          <div
            class="MuiGrid-root makeStyles-textContainer-4 MuiGrid-item"
          >
            <h1
              class="makeStyles-typography-3"
            >
              Connecting people with great ideas to web developers. Old website, new website, app, or feature, we've got you covered.
            </h1>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "mockSpies": Object {
    "setDarkMode": [MockFunction],
    "userDispatch": [MockFunction],
  },
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
